[JavaScript] Improve concept/boolean exercise (#1323)
exercism/v3: languages/javascript/exercises/concept/booleans/.docs/hints.md

### 1. Check if the 'Fast Attack' action is possible

- The logical NOT operator (`!`) can be placed before an expression to negate its value.

> - Check out [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Logical_NOT_!) on Logical Operators

Reviewer: Why is logical operators not capitalised on the point above, but is in this point?

Reviewer 2: Because the article name is Logical Operators.

> - If you'd like to dig deeper, you can check out [this article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators) by the awesome folks over at [MDN](https://developer.mozilla.org/en-US/)

Reviewer 2: ```suggestion
- Check out [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Logical_NOT_!) on Logical Operators
```

### 2. Check if the 'Approach and Spy' action is possible

- Logical expressions are evaluated from left to right and are tested for possible 'short-circuits'.
- For a more complete understanding of the left-to-right mechanic, you can take a look at [this article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence)!

### 3. Check if the 'Signal Prisoner' action is possible

- Logical operators in the order of their precedence (from highest to lowest): `!`, `&&`, `||`.

> -Check out [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Precedence_And_Associativity) on Operator Precedence

Reviewer 2: ```suggestion
- Check out [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Precedence_And_Associativity) on Operator Precedence
```

> - Check out [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Precedence_And_Associativity) on Operator Precedence

Reviewer: Same here. Does it need to be capitalised?

Reviewer 2: Same as above

> - If you'd like to dig deeper, you can check out [this article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators) by the awesome folks over at [MDN](https://developer.mozilla.org/en-US/)

Reviewer 2: We can simplify these:

```suggestion
-Check out [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Precedence_And_Associativity) on Operator Precedence
```

Digging deeper is a _great_ idea, but is more apt in `after.md`. These are really hints to get them to solve the exercise. If they need another hint, digging deeper is not an option :P
